source/Collection/CollectionView.js

```diff
@@ -177,6 +178,41 @@ export default class CollectionView extends React.PureComponent {
 
   /* ---------------------------- Component lifecycle methods ---------------------------- */
 
+  /**
+   * @private
+   * This method updates scrollLeft/scrollTop in state for the following conditions:
+   * 1) Empty content (0 rows or columns)
+   * 2) New scroll props overriding the current state
+   * 3) Cells-count or cells-size has changed, making previous scroll offsets invalid
+   */
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (
+      nextProps.cellCount === 0 &&
+      (prevState.scrollLeft !== 0 || prevState.scrollTop !== 0)
+    ) {
+      return {
+        scrollLeft: 0,
+        scrollTop: 0,
+      };
+    } else if (
+      nextProps.scrollLeft !== prevState.scrollLeft ||
+      nextProps.scrollTop !== prevState.scrollTop
+    ) {
+      return {
+        scrollLeft:
+          nextProps.scrollLeft != null
```

Comment:
nit: `!==` unless `nextProps` only contains updated props? since `undefined == null`

Reply:
This check handles null and undefined. It's probably not necessary but I'd prefer to be safe

Reply:
Ok that makes sense.